CRM/Financeextras/Hook/BuildForm/AdditionalPaymentButton.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+use CRM_Financeextras_ExtensionUtil as E;
+
+/**
+ * Implement Refund Button if supported Refund by Payment Processor
+ */
+class CRM_Financeextras_Hook_BuildForm__AdditionalPaymentButton {
+
+  /**
+   * @var string
+   *   Path where template with javascript stored for adding new Refund Button.
+   */
+  private $templatePath;
+
+  /**
+   * @var int
+   *   Contribution id for adding submit refund button.
+   */
+  private $contributionID;
+
+  /**
+   * @var \CRM_Contribute_Form_AdditionalPayment
+   *   Form object that is being altered.
+   */
+  private $form;
+
+  /**
+   * @var \Civi\Financeextras\Payment\Refund
+   */
+  private \Civi\Financeextras\Payment\Refund $paymentRefund;
+
+  /**
+   * CRM_Financeextras_Hook_BuildForm__AdditionalPaymentButton constructor.
+   *
+   * @param \CRM_Contribute_Form_AdditionalPayment $form
+   */
+  public function __construct(&$form) {
+    $this->form = $form;
+    $this->templatePath = E::path() . '/templates/CRM/AdditionalPayment/Form/Payment';
+    $this->contributionID = $this->form->getVar('_id');
+    $this->paymentRefund = new \Civi\Financeextras\Payment\Refund($this->contributionID);
+  }
+
+  /**
+   * Implements changes to form loaded in $this->form
+   */
+  public function buildForm() {
+    $this->addRefundButton();
+  }
+
+  /**
+   * Added Javascript in template to add the button for submit refund
+   */
+  private function addRefundButton() {
+    if (!$this->paymentRefund->contactHasRefundPermission()) {
+      return;
+    }
+
+    if (!$this->paymentRefund->isEligibleForRefund()) {
+      return;
+    }
+
+    $contribution_id = $this->contributionID;
+    $formButtonValues = [
+      "title" => ts('Submit Credit Card Refund'),
+      "icon" => "fa-chevron-right",
+      "link" => "financeextras/payment/refund/card?reset=1&id=" . $contribution_id,
```

Comment:
I think you can just use` $this->contributionID`

` "link" => "financeextras/payment/refund/card?reset=1&id=" . $this->contributionID,`